pytest_html/plugin.py

```diff
@@ -215,23 +218,51 @@ def pytest_sessionfinish(self, session):
             html.title('Test Report'),
             html.style(raw(style_css)))
 
+
+        class ResultByType:
+
+            def __init__(self, label="Passed", class_html="passed",
+                         data_type_associated="Passed", amount=0):
```

Comment:
I think we should have two required arguments: outcome and total. Then label, class, and data should be the same as the outcome unless they're provided.